### PR TITLE
fix: fixed some read-only mode buttons bugs

### PR DESCRIPTION
**Pull request by SystemChanger**
Fixes:
1. "setToolbarButtons()" now correctly caches buttons to "resizingDisabledButtons" array.
As it was: buttons were cached before being changed by this code:
```Javascript
let plugin, button, oldButton;
for (let key in pluginCallButtons) {
    if (!util.hasOwn(pluginCallButtons, key)) continue;
    plugin = plugins[key];
    button = pluginCallButtons[key];
    if (plugin.active && button) {
        oldButton = oldCallButtons[key];
        core.callPlugin(key, null, oldButton || button);
        if (oldButton) {
            button.parentElement.replaceChild(oldButton, button);
            pluginCallButtons[key] = oldButton;
        }
    }
}
```
As a result, they weren't disabled on the "readOnly()" invocations.

2. "setToolbarButtons()" now don't enable buttons disabled by "readOnly()".
As it was: On window resize (and other events with "setToolbarButtons()" invocation) buttons lost their "disabled" state.
(check the second point in "Suggestions" below)

3. Removed the "se-readonly-enabled" class from "Undo" and "Redo" buttons constructor.
As it was: they had this class, but anyway were "returned" in their handlers cause of readOnly. 
Obviously, they don't need them.

4. Now, read-only mode allows hiding more layer when the "more" button is pressed. Bonus: little code refactor.
As it was: the "more" button didn't react on pressing and hiding the layer in read-only mode.

Suggestions: (https://github.com/JiHong88/SunEditor/issues/835 issue)
1) Why "se-resizing-enabled" class and the corresponding array "resizingDisabledButtons" are called so?
This is super strange, They should be called something like "se-readonly-enabled" and "readOnlyDisabledButtons".
Version 3.0.0 ? :)
2) I think that it is better to copy all properties of previous toolbar buttons to the new ones in "setToolbarButtons()" method, cause in the future there could be some other stuff besides read-only, fullscreen or codeView (or is already there).  
Check the issue https://github.com/JiHong88/SunEditor/issues/835 for more details.

3) In addition, it'll be good to have "onSetToolbarButtons(buttonList)" user-definable event function to declare the custom buttons css and properties (ofc we can override this method, but it'll be cleaner for users with the knowledge that they have to do this for custom buttons). 
UPD. Created a separate pull request for "onSetToolbarButtons()" event:
https://github.com/JiHong88/SunEditor/pull/834

### CODE REVIEW

**Comment by SystemChanger** [comment]
We are getting close to the reason I started this all "read-only" adjustments ;)

File: src/lib/core.js [hunk 1]
```diff
@@ -7640,8 +7637,13 @@ export default function (context, pluginCallButtons, plugins, lang, options, _re
                 }
             }
 
+            core._cachingButtons();
+            core.history._resetCachingButton();
+
             if (core.hasFocus) event._applyTagEffects();
 
+            if (core.isReadOnly) util.setDisabledButtons(true, core.resizingDisabledButtons);
```
Comment (SystemChanger) [inline src/lib/core.js:7645]: There is no need in this line if you will apply the changes from https://github.com/JiHong88/SunEditor/pull/839 pull-request

**Comment by JiHong88** [comment]
@SystemChanger Thank you for your contribution.!
"se-resizing-enabled" is a variable created before "readonly" mode development.
These are buttons that are not deactivated when selecting images, etc.
It is also used in "readonly" mode.

**Comment by JiHong88** [comment]
@SystemChanger The [2.42.0](https://github.com/JiHong88/SunEditor/releases/tag/2.42.0) version has been updated.
If this issue has not been resolved, please reopen this issue.
Thank you.

**Pull request merged**